src/App.tsx

```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const selectValue = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [activeStart, setActiveStart] = useState(false);
+  const [valueSelect, setValueSelect] = useState(1);
+
+  return (
+    <div>
+      <div className={activeStart
+        ? 'App'
+        : 'App-start'}
+      >
+        <button
+          type="button"
+          className={activeStart
+            ? 'button is-danger is-light'
+            : 'button is-success is-light'}
+          onClick={() => {
+            return setActiveStart(!activeStart);
+          }}
+        >
+          {activeStart
+            ? 'Hide'
+            : 'Start'}
+        </button>
+
+        {activeStart
+          ? <GoodsList goods={goods} />
+          : null}
+
+        {activeStart
+          ? (
+            <>
+              <button
+                type="button"
+                className="button is-info is-light"
+                onClick={() => (
+                  setGoods([...goods].reverse())
```

Comment:
If you need to change the state depending on the previous state, you should use function inside setGoods
```suggestion
                  setGoods((prevGoods) => ([...prevGoods].reverse()))
```